content/concepts/protocols.md

```diff
@@ -1,227 +1,149 @@
-[multiplexed](/concepts/stream-multiplexing/). This is transparent to the protocol handler, which reads and writes
-unencrypted binary data over the stream.
+  - Eager writers can't flood readers
 
-The format of the binary data and the mechanics of what to send when and by whom are all up to the protocol to determine. For inspiration, some [common patterns](#common-patterns) that are used in libp2p's internal protocols are outlined below.
+Behind the scenes, libp2p will also ensure that the stream is 
+[secure](/concepts/secure-comms/) and efficiently
+[multiplexed](/concepts/stream-multiplexing/). This is transparent to the protocol 
+handler, which reads and writes unencrypted binary data over the stream.
 
+The protocol determines the binary data format and the transport mechanics. 
+For inspiration, some [common patterns](#common-patterns) that are used in libp2p's 
+internal protocols are outlined below.
 
 ## Protocol Negotiation
 
-When dialing out to initiate a new stream, libp2p will send the protocol id of the protocol you want to use.
-The listening peer on the other end will check the incoming protocol id against the registered protocol handlers.
-
-If the listening peer does not support the requested protocol, it will end the stream, and the dialing peer can
-try again with a different protocol, or possibly a fallback version of the initially requested protocol.
+When dialing out to initiate a new stream, libp2p sends the protocol ID of the 
+protocol you want to use. The listening peer on the other end checks the incoming 
+protocol ID against the registered protocol handlers.
 
-If the protocol is supported, the listening peer will echo back the protocol id as a signal that future data
-sent over the stream will
-use the agreed protocol semantics.
+If the listening peer does not support the requested protocol, it will send "na" on the stream
+and the dialing peer can try again with a different protocol or possibly a fallback 
+version of the initially requested protocol.
 
-This process of reaching agreement about what protocol to use for a given stream or connection is called
-**protocol negotiation**.
+If the protocol is supported, the listening peer will echo back the protocol ID as 
+a signal that future data sent over the stream will use the agreed protocol semantics.
 
+This process of reaching an agreement about what protocol to use for a given stream 
+or connection is called **protocol negotiation**.
 
-### Matching protocol ids and versions
+### Matching protocol IDs and versions
 
-When you register a protocol handler, there are two methods you can use.
+There are two methods available to register a protocol handler.
 
-The first takes two arguments: a protocol id, and a handler function. If an incoming stream request sends an exact
-match for the protocol id, the handler function will be invoked with the new stream as an argument.
+1. The first takes two arguments: a protocol ID and a handler function. If an incoming 
```

Comment:
This sounds more like the documentation for a specific API.

I don't think we use semver-matching anywhere in libp2p, so I'd prefer to keep the description _very_ short.

Reply:
We recently removed a sub-section on semver-matching in https://github.com/libp2p/docs/pull/185. Update the section to be generic but I'm not sure it is done in the best way